File: server/PQL-Domain-Question/src/main/java/com/passql/question/dto/QuestionDetail.java
package com.passql.question.dto;

import com.passql.question.constant.ChoiceKind;
import com.passql.question.constant.ChoiceSetSource;
import com.passql.question.constant.ChoiceSetStatus;
import com.passql.question.constant.ExecutionMode;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public record QuestionDetail(
    UUID questionUuid,
    String topicName,
    String subtopicName,
    Integer difficulty,
    ExecutionMode executionMode,
    String stem,
    String schemaDisplay,
    String schemaDdl,
    String schemaSampleData,
    String schemaIntent,
    String answerSql,
    String hint,
    List<ChoiceSetSummary> choiceSets
) {
    public record ChoiceSetSummary(
        UUID choiceSetUuid,
        ChoiceSetSource source,
        ChoiceSetStatus status,
        Boolean sandboxValidationPassed,
        LocalDateTime createdAt,
        List<ChoiceItem> items
    ) {}

    public record ChoiceItem(String key, ChoiceKind kind, String body, Boolean isCorrect, String rationale, Integer sortOrder) {}
}

File: lib/data/models/question/question_detail.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'choice_set_summary.dart';

part 'question_detail.freezed.dart';
part 'question_detail.g.dart';

/// 문제 상세 응답. executionMode에 따라 UI 분기.
/// EXECUTABLE: SQL 실행 + 스키마 표시
/// CONCEPT_ONLY: 텍스트 선택지만 표시
@freezed
class QuestionDetail with _$QuestionDetail {
  const factory QuestionDetail({
    required String questionUuid,
    String? topicName,
    String? subtopicName,
    int? difficulty,
    String? executionMode,
    required String stem,
    String? schemaDisplay,
    String? schemaDdl,
    String? schemaSampleData,
    String? schemaIntent,
    String? answerSql,
    String? hint,
    @Default([]) List<ChoiceSetSummary> choiceSets,
  }) = _QuestionDetail;

  factory QuestionDetail.fromJson(Map<String, dynamic> json) =>
      _$QuestionDetailFromJson(json);
}
